src/components/LineChart/LineChart.jsx

```diff
@@ -311,75 +312,87 @@ class LineChart extends PureComponent {
         const goodYmax = yScale(incidentData[selectedASN][incIndex].goodPeriodMetric);
         const badYmax = yScale(incidentData[selectedASN][incIndex].badPeriodMetric);
 
+        const goodStart = incidentData[selectedASN][incIndex].goodPeriodStart;
+        const goodEnd = incidentData[selectedASN][incIndex].goodPeriodEnd;
+        const badStart = incidentData[selectedASN][incIndex].badPeriodStart;
+        const badEnd = incidentData[selectedASN][incIndex].badPeriodEnd;
+
         const goodWidth = xScale(incidentData[selectedASN][incIndex].goodPeriodEnd)
           - xScale(incidentData[selectedASN][incIndex].goodPeriodStart);
         const badWidth = xScale(incidentData[selectedASN][incIndex].badPeriodEnd)
           - xScale(incidentData[selectedASN][incIndex].badPeriodStart);
         const goodHeight = plotAreaHeight - goodYmax;
-        const badHeight = plotAreaHeight - badYmax;
-        const incidentHeight = Math.abs(badYmax - goodYmax);
-        // NOTE: This also must be manually tuned. It hides hover text in the case that the area is too small for the text to fit.
-        const rectFitsText = (goodWidth > 180) && (badWidth > 180);
+        // const badHeight = plotAreaHeight - badYmax;
+        // const incidentHeight = Math.abs(badYmax - goodYmax);
+        // Values chosen for aesthetic purposes
+        const dx = 30;
+        const dy = 20;
+        const padding = 10;
+
+        const combinedData = [...series, ...annotationSeries];
+        let xDomain = xExtent;
+        if (!xDomain) {
+          xDomain = multiExtent(combinedData, d => d[xKey], oneSeries => oneSeries.results);
+        }
+        let yDomain = yExtent;
+        if (!yDomain) {
+          yDomain = multiExtent(combinedData, d => d[yKey], oneSeries => oneSeries.results) || [];
 
-        // TODO(amy): fix linter issues here without breaking hover functionality
-        // Draw the hover state for the good period information
-        if (highlightedDate.isBefore(incidentData[selectedASN][incIndex].goodPeriodEnd) && highlightedDate.isSameOrAfter(incidentData[selectedASN][incIndex].goodPeriodStart) && mouseY > goodYmax) {
-          this.infoHoverBox.append('rect')
-          .classed('good-incident-area', true)
-          .attr('x', xScale(incidentData[selectedASN][incIndex].goodPeriodStart))
-          .attr('y', goodYmax)
-          .attr('width', goodWidth)
-          .attr('height', plotAreaHeight - goodYmax);
-
-          if (rectFitsText) {
-            this.infoHoverBox.append('text')
-            .classed('good-hover-text', true)
-            .attr('x', xScale(incidentData[selectedASN][incIndex].goodPeriodStart) + goodWidth / 2)
-            .attr('y', goodYmax + goodHeight / 2)
-            .attr('alignment-baseline', 'central')
-            .attr('text-anchor", "middle')
-            .text(goodDescription)
+          if (yDomain[0] == null) {
```

Comment:
Can you make this === instead? (it represents strict equality)

Reply:
and in all other places where you use ==

Reply:
Changed!